### PR TITLE
Fixes mapping of active statement when changing ctor body from block to expression

### PR DESCRIPTION
**Pull request by tmat**
### Customer scenario

VS crashes when program is stopped at a breakpoint placed on the constructor declaration (not on its body) and the constructor body is changed from a block to an expression.

Harden GetAllAddedSymbolsAsync to avoid VS crash on an unexpected exception.

### Bugs this fixes

[VSO 590387](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/590387)

### Workarounds, if any

None.

### Risk

Small.

### Performance impact

Small.

### Root cause analysis

Missing unit test.

### How was the bug found?

Customer reported.

### CODE REVIEW

**Comment by tmat** [comment]
@ivanbasov I have implemented additional fixes to GetAllAddedSymbolsAsync method that can only be done in -vs-deps branch.

**Comment by tmat** [comment]
@dotnet/roslyn-compiler Please review a simple addition to `FatalError` helpers.

**Comment by tmat** [comment]
@jinujoseph 15.7 bug

File: src/Features/Core/Portable/EditAndContinue/EditSession.cs [hunk 1]
```diff
@@ -281,33 +281,47 @@ private List<(DocumentId, AsyncLazy<DocumentAnalysisResults>)> GetChangedDocumen
...
+                HashSet<ISymbol> addedSymbols = null;
+                foreach (var analysis in analyses)
                 {
-                    if (edit.Kind == SemanticEditKind.Insert)
+                    // Only consider analyses for documents that belong the currently analyzed project.
+                    if (analysis.Document.Project == project)
                     {
-                        if (addedSymbols == null)
+                        var results = await analysis.Results.GetValueAsync(cancellationToken).ConfigureAwait(false);
+                        if (!results.HasChangesAndErrors)
```
Comment (ivanbasov) [inline src/Features/Core/Portable/EditAndContinue/EditSession.cs]: >if (!results.HasChangesAndErrors) [](start = 24, length = 33)

I believe you intentionally just skipping analysis providing errors not making a complete bail out.
Maybe a comment would be helpful.

**Comment by tmat** [comment]
test windows_debug_vs-integration_prtest please

**Comment by jinujoseph** [comment]
approved to merge via [link](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/590387) for 15.7.Preview5

**Pull request merged**